prime-router/src/main/kotlin/cli/tests/TestReportStream.kt

```diff
@@ -816,16 +824,19 @@ abstract class CoolTest {
               from item_lineage as IL
               join report_file as RF on IL.child_report_id = RF.report_id
               join action as A on A.action_id = RF.action_id
-              where RF.receiving_org_svc = ?
-              ${if (receivingOrg != null) "and RF.receiving_org = ?" else ""}
-              and A.action_name = ?
+              where
+              ${if (receivingOrgSvc != null) "RF.receiving_org_svc = ? and" else ""}
+              ${if (receivingOrg != null) "and RF.receiving_org = ? and" else ""}
+              A.action_name = ?
```

Comment:
These changes feel like a mess, I'd love suggestions of a better way.

Reply:
Other than separating it into multiple queries depending on what is passed in I've got nothing. I don't think that would actually help the situation to do that - this one may just be a (functional) mess.